src/pykeen/experiments/nodepiece/galkin2022_nodepiece_fb15k237.yaml

```diff
@@ -0,0 +1,43 @@
+metadata:
+  title: Learn FB15k-237 Dataset with NodePiece + RotatE as described by Galkin et al., 2022
+pipeline:
+  dataset: fb15k237
+  evaluator_kwargs:
+    filtered: true
+  loss: BCEWithLogitsLoss
+  model: NodePiece
+  model_kwargs:
+    entity_initializer: xavier_uniform_
+    relation_initializer: init_phases
+    relation_constrainer: complex_normalize
```

Comment:
since the representations are already instantiated inside the `NodePiece` part
https://github.com/pykeen/pykeen/blob/ecc2c22e6e4edbeb6700e4af4969d99d925df9eb/src/pykeen/models/unimodal/node_piece.py#L166-L185

they should be passed there, right?

Reply:
Or actually be part of the embedding specification?
https://github.com/pykeen/pykeen/blob/ecc2c22e6e4edbeb6700e4af4969d99d925df9eb/src/pykeen/models/unimodal/node_piece.py#L149

Reply:
Right, it seems that `EmbeddingSpecification`  is the correct place to put all initializers, regularizers and constrainers. The catch is to make sure it's possible to initialize tokens and relations with different parameters - right now it seems that both tokens and relations are created from the same specification 🤔

Reply:
I think you would need to implement this logic inside the `NodePiece` model in `pykeen.models.unimodal`, so it can be used via the pipeline.

Reply:
Solved that in the `NodePiece` model by adding
```python
        anchor_specification = embedding_specification or EmbeddingSpecification(
            shape=(embedding_dim,),
            initializer=kwargs.pop('entity_initializer', None),
            normalizer=kwargs.pop('entity_normalizer', None),
            constrainer=kwargs.pop('entity_constrainer', None),
            regularizer=kwargs.pop('entity_regularizer', None),
        )
        relation_specification = EmbeddingSpecification(
            shape=(embedding_dim,),
            initializer=kwargs.pop('relation_initializer', None),
            normalizer=kwargs.pop('relation_normalizer', None),
            constrainer=kwargs.pop('relation_constrainer', None),
            regularizer=kwargs.pop('relation_regularizer', None)
        )
```

This part needs to be added to both NodePiece and InductiveNodePiece, will do that after we merge the main InductiveLP PR

Reply:
You could also directly add them to the `__init__` signature with appropriate default value, e.g., `relation_initializer: Hint[Initializer = None]`.

Reply:
finally done here https://github.com/pykeen/pykeen/pull/771/commits/b3ed8c942200f1ddde9c33dd54d1443c15f885a4